src/install/lock.ts

```diff
@@ -41,15 +35,15 @@ export interface LockResolutions {
   };
 }
 
+interface PackageTargetMap {
+  [pkgName: string]: PackageTarget | URL,
```

Comment:
This is the interface actually used by the code - I've fixed all the type issues related to `VersionConstraints`, it was causing some weird bugs where the type doesn't match the actual input.